src/NBitcoin/ChainBase.cs

```diff
@@ -104,7 +104,7 @@ public bool TrySetTip(BlockHeader header, out ChainedBlock chainedHeader)
             if (prev == null)
                 return false;
 
-            chainedHeader = new ChainedBlock(header, header.GetHash(), this.GetBlock(header.HashPrevBlock));
+            chainedHeader = new ChainedBlock(header, header.GetHash(NetworkOptions.TemporaryOptions), this.GetBlock(header.HashPrevBlock));
```

Comment:
This is going to break to node as `TrySetTip` is called from `SetTip` which is called from the ChainHeaderBehaviour and is used bot in POS and POW meaning we will be using a POW Options.

We probably need to pass Options in the method, or set it somehow in ChainedBlock.

Reply:
I would suggest you try to sync stratis with this code to make sure I am right.

Reply:
See my other comment if we pass Network to ChainBase constructor then we are all good.